lib/input/sftp.go

```diff
@@ -200,6 +250,9 @@ func (s *sftpReader) ReadWithContext(ctx context.Context) (types.Message, reader
 			err = types.ErrTimeout
 		}
 		if err != types.ErrTimeout {
+			if s.conf.Watcher.Enabled {
+				err = s.watcherCache.Set(s.currentPath, []byte{})
```

Comment:
Rather than `[]byte{}` we probably need to set a real >0 byte value as some cache types might not allow an empty value. I'd recommend just using a single char like `[]byte("@")`.